org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/DOMUtils.java

```diff
@@ -134,4 +139,25 @@ public static DOMDocument loadDocument(String documentURI, URIResolverExtensionM
 			return null;
 		}
 	}
+
+	/**
+	 * Returns an instance of SAX parser factory by disabling external entities
+	 * declarations.
+	 * 
+	 * @return an instance of SAX parser factory by disabling external entities
+	 *         declarations.
+	 * @throws SAXNotRecognizedException
+	 * @throws SAXNotSupportedException
+	 * @throws ParserConfigurationException
+	 */
+	public static SAXParserFactory newSAXParserFactory()
+			throws SAXNotRecognizedException, SAXNotSupportedException, ParserConfigurationException {
+		SAXParserFactory factory = SAXParserFactory.newInstance();
+		// to be compliant, completely disable DOCTYPE declaration:
+		factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+		// or completely disable external entities declarations:
+		factory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+		factory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);

```

Comment:
I think this is unnecessary once you disabled DOCTYPE declaration

Reply:
Ok I fixed it.